www/js/controllers.js

```diff
@@ -75,6 +76,43 @@ angular.module('slowfood.controllers', [])
         };
     };
 
+    $scope.pickMenu = function(index) {
+        $ionicModal.fromTemplateUrl('templates/show-menu.html', {
+            scope: $scope,
+            animation: 'slide-in-up'
+        }).then(function (modal) {
+            $scope.modal = modal;
+            $scope.currentMenuIndex = index;
+            $scope.getCategoryArrays(index);
+            $scope.openModal();
+        });
+    };
+
+    $scope.getCategoryArrays = function(index) {
+        var startersArray = [];
+        var mainsArray = [];
+        var dessertsArray = [];
+        var menu = restaurant_response.menus[index];
+        for (i = 0; i < menu.dishes.length; i++) {
+            switch(menu.dishes[i].category) {
+                case 'Starters':
+                    startersArray.push(menu.dishes[i]);
+                    break;
+                case 'Mains':
+                    mainsArray.push(menu.dishes[i]);
+                    break;
+                case 'Desserts':
+                    dessertsArray.push(menu.dishes[i]);
+                    break;
+                default:
+                    return true;
+            }
+        }
+        $scope.startersArray = startersArray;
+        $scope.mainsArray = mainsArray;
+        $scope.dessertsArray = dessertsArray;
+    };
+
```

Comment:
Creative.
I'll review this at home tonight. I just might have a slightly different solution for you.